Remove custom `Icon` component in favour of `@wordpress/icons` where possible (#5599)
woocommerce/woocommerce-gutenberg-products-block: assets/js/icons/README.md

## Usage

> Note we use hte `Icon` component from `@wordpress/icons`, and the Icon SVG from `@woocommerce/icons`.

Reviewer: ```suggestion
Note we use the `Icon` component from `@wordpress/icons`. We use some SVG icons from `@woocommerce/icons` for WC Blocks specific icons, but we also use existing icons from `@wordpress/icons`.
```

```js
import { woo } from '@woocommerce/icons';

> import { Icon } from '@wordpress/icons';

Reviewer: ```suggestion
import { Icon, postComments } from '@wordpress/icons';
```

> <Icon icon={ woo } />

Reviewer: ```suggestion
<Icon icon={ woo } /> // icon  from '@woocommerce/icons'
<Icon icon={ postComments } /> // icon from '@wordpress/icons'
```

## Usage

```js

> import { woo } from '@woocommerce/icons';

Reviewer: It would be great if we had a small comment explaining why the two imports. 

Also, under `## Adding Icons` we could mention

0. Make sure the icon is not already included in [the Library](https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library) that comes with @wordpress/icons package.
